src/OutputWriters/output_construction.jl

```diff
@@ -16,6 +17,24 @@ end
 ##### Function output fallback
 #####
 
+function mask_field!(output)
+    field_masking_event = mask_immersed_field!(output)
+    wait(field_masking_event)
+end
+
+
+function _construct_output(output, grid, kwargs...)
+    if output isa AbstractOperation
+        output = Field(output)
```

Comment:
This will allocate memory for an abstract operation output _every time_ output is written. We don't want this --- instead, we want to allocate memory when the output writer is constructed.

Reply:
Sorry, I thought that the usage of `construct_output` was changed, but I see now that it's not so I revoke this comment. But there are still changes needed below.

Reply:
Sounds good. I'll mark this as resolved